Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm

```diff
@@ -620,34 +620,36 @@ - (BOOL)prefersStatusBarHidden
 
     interface->setVideoPresentationModel(model.ptr());
 
-    return std::make_tuple(WTFMove(model), WTFMove(interface));
+    return std::make_tuple(WTFMove(model), interface.releaseNonNull());
 }
 
-VideoPresentationManagerProxy::ModelInterfaceTuple& VideoPresentationManagerProxy::ensureModelAndInterface(PlaybackSessionContextIdentifier contextId)
+const VideoPresentationManagerProxy::ModelInterfaceTuple& VideoPresentationManagerProxy::ensureModelAndInterface(PlaybackSessionContextIdentifier contextId)
 {
-    auto addResult = m_contextMap.add(contextId, ModelInterfaceTuple());
-    if (addResult.isNewEntry)
-        addResult.iterator->value = createModelAndInterface(contextId);
+    auto it = m_contextMap.find(contextId);
+    if (it != m_contextMap.end())
+        return it->value;
+    auto addResult = m_contextMap.add(contextId, createModelAndInterface(contextId));
```

Comment:
This new code is less efficient, why not using m_contextMap.ensure instead?

Reply:
It’s only less efficient when creating an entry. It’s just as quick for the main use (an entry already exists). 

With the tuple now taking a ref, they must be created immediately, we can no longer use the previous paradigm

Reply:
ok, I used `HashMap::ensure` instead.